src/rred_reports/reports/interface.py

```diff
@@ -0,0 +1,99 @@
+from enum import Enum
+from pathlib import Path
+from typing import Optional
+
+import pandas as pd
+import tomli
+import typer
+from loguru import logger
+
+from rred_reports.reports.generate import generate_report_school
+
+app = typer.Typer()
+
+
+class ReportType(str, Enum):
+    """ReportType class
+
+    Provides report types as enums
+    """
+
+    SCHOOL = "school"
+    CENTRE = "centre"
+    NATIONAL = "national"
+    ALL = "all"
+
+
+def validate_data_sources(year: int, template_file: Path, top_level_dir: Optional[Path] = None) -> dict:
+    """Perform some basic data source validation
+    Args:
+        year (int): Year to process
+        template_file_path (Path): Path to template file
+
+    Returns:
+        dict: Dictionary of validated data sources
+    """
+    if top_level_dir is None:
+        top_level_dir = Path(__file__).resolve().parents[3]
+
+    data_path = top_level_dir / "output" / "processed" / str(year)
+    template_file_path = top_level_dir / template_file
+    output_dir = top_level_dir / "output" / "reports" / str(year) / "schools"
+
+    try:
+        processed_data = pd.read_csv(data_path / "processed_data.csv")
+    except FileNotFoundError as error:
+        logger.error(f'No processed data file found at {data_path / "processed_data.csv"}. Exiting.')
+        raise error
+
+    try:
+        assert template_file_path.is_file()
+    except AssertionError as error:
+        logger.error(f"No template file found at {template_file_path}. Exiting.")
+        raise FileNotFoundError from error
+
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True)
+
+    return {"data": processed_data, "template_file": template_file_path, "output_dir": output_dir}
+
+
+@app.command()
+def create(level: ReportType, year: int, config_file: str = "report_config.toml"):
+    """Generate a report at the level specified
+
+    Args:
+        level (ReportType): school, centre, national
+        year (int): Year to process
+    """
+    typer.echo(f"Creating a report for level: {level.value}")
+    config = get_config(Path(__file__) / config_file) if config_file == "report_config.toml" else get_config(config_file)
```

Comment:
Would we be able to just have the `config_file` defined as `config_file: Path = Path("report_config.toml")`
and then we could simplify this as the path to the file will be given as relative anyway? I could be wrong about that though I'm relatively sure typer makes this nice
```suggestion
    config = get_config(config_file)
```

Reply:
Happy to try this out, wasn't sure how the relative paths would shake out or which directory we'd want to pop the report config toml in

Reply:
Ah I remember this was set as a string at least, `ruff` shouts at you with
```zsh
src/rred_reports/reports/interface.py:69:62: B008 Do not perform function call `Path` in argument defaults
```

Could still do as you've suggested above but cast it as `Path` in the call to `get_config`?

Reply:
Ah okay, got it - think this is resolved now, have tested with and without the `--config-file` argument and it picks up the config file in both cases